src/backend.md

```diff
@@ -486,6 +486,57 @@ One of the key principles in effective transaction management is the avoidance o
    }
    ```
 
+1. The set of a spray job preconditions must be greater than the set of an action job preconditions. They must have an intersection as large as possible.
```

Comment:
> The set of a spray job preconditions must be greater than the set of an action job preconditions.

It is not possible for some cases. 

For example, please check the `createTicketsByCompanyEvents` job in Roberto:
https://github.com/OsomePteLtd/roberto/blob/main/src/jobs/companyEvent/createTicketsByCompanyEvents.job.ts

It finds a set of companyEvents potentially ready for creating work tickets, but the main logic lives inside companyEvent adapters. I don't think that we should overcomplicate the query in `getNextBatch` for this case.

Reply:
I am not sure we need to make a principle from any common sense. Sure we should strive to fire fewer events, but the rule above has too many exclusions to be a principle IMHO.